pkg/chains/chains.go

```diff
@@ -368,14 +376,47 @@ func ZetaChainFromChainID(chainID string) (Chain, error) {
 	case ZetaChainDevnet.ChainId:
 		return ZetaChainDevnet, nil
 	default:
-		return Chain{}, fmt.Errorf("chain %d not found", ethChainID)
+		return Chain{}, ErrNotZetaChain
+	}
+}
+
+// CombineDefaultChainsList combines the default chains list with a list of chains
+// duplicated chain ID are overwritten by the second list
+func CombineDefaultChainsList(chains []Chain) []*Chain {
+	chainsPtr := ChainListPointers(chains)
+
+	return CombineChainList(DefaultChainsList(), chainsPtr)
+}
+
+// CombineChainList combines a list of chains with a list of chains
+// duplicated chain ID are overwritten by the second list
+func CombineChainList(chains1 []*Chain, chains2 []*Chain) []*Chain {
+	combined := make([]*Chain, 0)
+	combined = append(combined, chains1...)
+
+	// map chain ID in chains1 to index in the list
```

Comment:
Seems the comment doesn't match the loop below. The loop below is not iterating on `chains1`.